configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java

```diff
@@ -141,6 +141,23 @@ public void createAndPrepareAndActivate() {
         assertTrue(result.configChangeActions().getRestartActions().isEmpty());
     }
 
+    @Test
+    public void redeploy() {
+        PrepareResult result = deployApp(testApp);
+
+        long firstSessionId = result.sessionId();
+
+        PrepareResult result2 = deployApp(testApp);
+        long secondSessionId = result2.sessionId();
+        assertNotEquals(firstSessionId, secondSessionId);
+
+        TenantName tenantName = applicationId().tenant();
+        Tenant tenant = tenantRepository.getTenant(tenantName);
+        LocalSession session = tenant.getLocalSessionRepo().getSession(
+                tenant.getApplicationRepo().requireActiveSessionOf(applicationId()));
+        assertEquals(firstSessionId, session.getMetaData().getPreviousActiveGeneration());
```

Comment:
Perhaps we should consider to stop using 'generation' if it really is the sessionId.

Reply:
Agreed, it's just an implementation detail that they are equal. I'll try to improve this in a later PR.